react-hot-button-component/src/index.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+let counter = 1;
+
+class CustomButton extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {isClicked: false};
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    
+
+    handleClick() {
+        this.setState({isClicked: counter++})
+    }
+
+    render() {
+        if(this.state.isClicked >= 18){
+            return <button className='hot6' onClick={this.handleClick}>Hot Button</button>
+        }
+        if(this.state.isClicked >= 15){
+            return <button className='hot5' onClick={this.handleClick}>Hot Button</button>
+        }
+        if(this.state.isClicked >= 12){
+            return <button className='hot4' onClick={this.handleClick}>Hot Button</button>
+        }
+        if(this.state.isClicked >= 9){
+            return <button className='hot3' onClick={this.handleClick}>Hot Button</button>
+        }
+        if(this.state.isClicked >= 6){
+            return <button className='hot2' onClick={this.handleClick}>Hot Button</button>
+        }
+        if(this.state.isClicked >= 3){
+            return <button className='hot1' onClick={this.handleClick}>Hot Button</button>
+            
+        }
+        return <button className='hot' onClick={this.handleClick}>Hot Button</button>
+    }

```

Comment:
We want to avoid a bunch of duplication of effort. The only thing that changes between these buttons is the class name. So it is better to create 1 button and just dynamically set the class name. For example:

```JSX
render() {
  let className = 'hot';

  if(this.state.isClicked >= 18){
    className = 'hot6';
  }
  if(this.state.isClicked >= 15){
    className = 'hot5';
  }
  if(this.state.isClicked >= 12){
    className = 'hot4';
  }
  if(this.state.isClicked >= 9){
    className = 'hot3';
  }
  if(this.state.isClicked >= 6){
    className = 'hot2';
  }
  if(this.state.isClicked >= 3){
    className = 'hot1';
  }
  
  return <button className={className} onClick={this.handleClick}>Hot Button</button>
}